packages/flutter_tools/lib/src/run.dart

```diff
@@ -471,7 +477,15 @@ class DevFS {
       List<_DevFSFile> files = toSend.map((_DevFSFileEntry entry) {
         return new _DevFSFile('/${entry.devPath}', entry.file);
       }).toList();
-      await observatory.writeDevFSFiles(fsName, files: files);
+
+      // TODO(devoncarew): Batch this up in larger groups using writeDevFSFiles().
+      await Future.wait(files.map((_DevFSFile file) {
+        return observatory.writeDevFSFile(
+          fsName,
+          path: file.path,
+          fileContents: file.getContents()
+        );
+      })).timeout(new Duration(seconds: 10));
```

Comment:
I'll add a note here to callout the fact that there will be dangling calls. This will be re-written in a bit to batch up sending files into larger groups.